### ISSUE TITLE
pmml export from tf.estimator models

### ISSUE THREAD

**Issue by Debasish-Das-CK** [jpmml/jpmml-tensorflow#9]
I built a model with tf.estimator API (dnn and linear). I can load it up on python serving by doing the following steps:
```
import tensorflow as tf
from tensorflow.python.saved_model.tag_constants import tag_constants
with tf.Session(graph=tf.Graph()) as sess:
  tf.saved_model.loader.load(sess, [tag_constants.TRAINING], export_dir)
```

with jpmml-tensorflow I am getting a factory issue:
```
cmle-env) debasish:jpmml-tensorflow debasish.das$ java -jar target/converter-executable-1.0-SNAPSHOT.jar --tf-savedmodel-input dnn/ --pmml-output dnn.pmml
Mar 22, 2018 5:30:25 PM org.jpmml.tensorflow.Main run
INFO: Parsing SavedModel..
2018-03-22 17:30:25.707794: I tensorflow/core/platform/cpu_feature_guard.cc:140] Your CPU supports instructions that this TensorFlow binary was not compiled to use: SSE4.2 AVX AVX2 FMA
2018-03-22 17:30:25.708049: I tensorflow/cc/saved_model/loader.cc:242] Loading SavedModel with tags: { serve }; from: /Users/debasish.das/creditkarma/jpmml-tensorflow/dnn
2018-03-22 17:30:25.802902: I tensorflow/cc/saved_model/loader.cc:161] Restoring SavedModel bundle.
2018-03-22 17:30:25.841061: I tensorflow/cc/saved_model/loader.cc:196] Running LegacyInitOp on SavedModel bundle.
2018-03-22 17:30:25.890051: I tensorflow/cc/saved_model/loader.cc:291] SavedModel load for tags { serve }; Status: success. Took 182028 microseconds.
Mar 22, 2018 5:30:26 PM org.jpmml.tensorflow.Main run
INFO: Parsed SavedModel in 705 ms.
Mar 22, 2018 5:30:26 PM org.jpmml.tensorflow.Main run
INFO: Converting..
Mar 22, 2018 5:30:26 PM org.jpmml.tensorflow.Main run
SEVERE: Failed to convert
java.lang.IllegalArgumentException
	at org.jpmml.tensorflow.EstimatorFactory.newEstimator(EstimatorFactory.java:58)
	at org.jpmml.tensorflow.Main.run(Main.java:117)
	at org.jpmml.tensorflow.Main.main(Main.java:90)

Exception in thread "main" java.lang.IllegalArgumentException
	at org.jpmml.tensorflow.EstimatorFactory.newEstimator(EstimatorFactory.java:58)
	at org.jpmml.tensorflow.Main.run(Main.java:117)
	at org.jpmml.tensorflow.Main.main(Main.java:90)
```
Do I need to add a new Estimator for the tf.estimator models ?

**Comment by Debasish-Das-CK** [comment]
I looked in nodeMap.keySet() in the EstimatorFactory...looks like there is no string like  "dnn/binary_logistic_head/predictions/probabilities"...perhaps the tf.estimator.DNNClassifier has changed from tf.contrib.learn DNNClassifier

**Comment by Debasish-Das-CK** [comment]
dnn/head/predictions/logits/Shape, dnn/head/predictions/logits/assert_rank_at_least/rank, dnn/head/predictions/logits/assert_rank_at_least/assert_type/statically_determined_correct_type, dnn/head/predictions/logits/assert_rank_at_least/static_checks_determined_all_ok, dnn/head/predictions/logistic, dnn/head/predictions/zeros_like, dnn/head/predictions/two_class_logits/axis, dnn/head/predictions/two_class_logits, dnn/head/predictions/probabilities, dnn/head/predictions/class_ids/dimension, dnn/head/predictions/class_ids, dnn/head/predictions/ExpandDims/dim, dnn/head/predictions/ExpandDims, dnn/head/predictions/str_classes

These are the names I found based dnn/head/predictions...none of them matches with dnn/binary_logistic_head

**Comment by vruusmann** [comment]
>  There is no string like "dnn/binary_logistic_head/predictions/probabilities"

This "model type" hint string is devised and set by the `tensorflow.contrib.learn.DNNClassifier` class. If you use any other TensorFlow wrapper APIs/classes (or don't use any wrapper at all), you shall be getting different "model type" hint strings in your SavedModel directories.

For example, this is how I trained and exported my sample SavedModel directories about ~1 year ago:
https://github.com/jpmml/jpmml-tensorflow/blob/master/src/test/resources/main.py

Maybe you'll be getting different "model type" hints if you run this script today. The lack of conventions/common identifiers/common data structures makes TensorFlow conversion extremely difficult.

The path forward would be to use proxy data formats such as DL4J or ONNX. In other words, the conversion workflow shall be `TensorFlow -> (DL4J | ONNX) -> PMML` (instead of `TensorFlow -> PMML`).

**Comment by Debasish-Das-CK** [comment]
sounds good...we will stick with tensorflow serving...
